qiime2/sdk/plugin_manager.py

```diff
@@ -101,43 +122,112 @@ def _integrate_plugin(self, plugin):
+            The semantic type is used to filter the formats associated with
+            that specific semantic type
 
+        This method will filter out the formats using the filter provided by
+        the user and the semantic type. The return is a set of filtered
+        formats.
         """
-        importable_formats = {}
-        for name, record in self.formats.items():
-            from_type = transform.ModelType.from_view_type(
-                record.format)
+        if filter is not None and filter not in GetFormatFilters:
+            raise ValueError("The format filter provided: %s is not "
+                             "valid.", (filter))
+
+        if semantic_type is None:
+            formats = set(f.format for f in self.type_formats)
+
+        else:
+            formats = set()
+
+            if isinstance(semantic_type, str):
+                semantic_type = parse_type(semantic_type, "semantic")
+
             for type_format in self.type_formats:
-                to_type = transform.ModelType.from_view_type(
-                    type_format.format)
-                if from_type.has_transformation(to_type):
-                    importable_formats[name] = record
+                if semantic_type <= type_format.type_expression:
+                    formats.add(type_format.format)
                     break
-        return importable_formats
 
-    @property
-    def importable_types(self):
-        """Return set of concrete semantic types that are importable.
+            if not formats:
+                raise ValueError("No formats associated with the type "
+                                 f"{semantic_type}.")
+
+        transformable_formats = set(formats)
+
+        if filter is None or GetFormatFilters.IMPORTABLE in filter:
+            transformable_formats.update(
+                self._get_formats_helper(formats, self._reverse_transformers))
+
+        if filter is None or GetFormatFilters.EXPORTABLE in filter:
+            transformable_formats.update(
+                self._get_formats_helper(formats, self.transformers))
 
-        A concrete semantic type is importable if it has an associated
-        directory format.
+        result_formats = {}
+        for format_ in transformable_formats:
+            format_ = format_.__name__
+            result_formats[format_] = self.formats[format_]
 
+        return result_formats
+
+    def _get_formats_helper(self, formats, transformer_dict):
         """
-        importable_types = set()
-        for type_format in self.type_formats:
-            for type in type_format.type_expression:
-                importable_types.add(type)
-        return importable_types
+        _get_formats_helper(self, formats, transformer_dict)
+
+        formats : set(DirectoryFormat)
+            We are finding all formats that can be directly transformed to and
+            from formats in this set
+
+        tranformer_dict : Class: Dict{ Class: TransformerRecord }
```

Comment:
`Dict[str, Dict[str, TransformerRecord]]` (I think it's a str anyhow, update if not)